src/common/servers/serverManager.ts

```diff
@@ -313,10 +321,8 @@ export class ServerManager extends EventEmitter {
         // Set the current server based on config if the user last used a local server
         // Otherwise, just use the first server
         let currentServerIndex = Config.lastActiveServer ?? 0;
-        if (Config.localServers.length && Config.predefinedServers.length) {
-            currentServerIndex += Config.predefinedServers.length;
-        } else if (Config.predefinedServers.length) {
-            currentServerIndex = 0;
+        if (currentServerIndex >= initialServers.length - 1) {
```

Comment:
Should this be ">=" and "-1"?  Feels like an off by one.

Reply:
Also, the PR description suggests "Config.lastActiveServer" can be -1.   In that case the conditional equates to false, and currentServerIndex remains -1.

Reply:
I should clarify: `Config.lastActiveServer` CANNOT be -1. See: https://github.com/mattermost/desktop/blob/723c626b55ddd72a49958b087c2892ba5d93eda5/src/common/Validator.ts#L179

The validation would fail and (what I neglected to mention) the config would be reset and the value would be `0`.